ros/processor/insights_engine_consumer.py

```diff
@@ -82,31 +87,41 @@ def run(self):
                 self.consumer.commit()
 
     def handle_msg(self, msg):
-        if system_allowed_in_ros(msg, self.reporter):
-            host = msg["input"]["host"]
-            platform_metadata = msg["input"]["platform_metadata"]
-            system_metadata = msg["results"]["system"]["metadata"]
-
-            threadctx.request_id = platform_metadata.get('request_id')
-            threadctx.account = platform_metadata.get('account')
-            threadctx.org_id = platform_metadata.get('org_id')
-
-            performance_record = get_performance_profile(
-                platform_metadata['url'],
-                platform_metadata.get('org_id'),
-                host['id'],
-                custom_prefix=self.prefix
-            )
-
-            reports = []
-            if msg["results"]["reports"] \
-                    and isinstance(msg["results"]["reports"], list):
-                reports = msg["results"]["reports"]
-            ros_reports = [
-                report for report in reports
-                if 'ros_instance_evaluation' in report["rule_id"]
-            ]
-            self.process_report(host, platform_metadata, ros_reports, system_metadata, performance_record)
+        with app.app_context():
+            if system_allowed_in_ros(msg, self.reporter):
+                host = msg["input"]["host"]
+                platform_metadata = msg["input"]["platform_metadata"]
+                system_metadata = msg["results"]["system"]["metadata"]
+
+                threadctx.request_id = platform_metadata.get('request_id')
+                threadctx.account = platform_metadata.get('account')
+                threadctx.org_id = platform_metadata.get('org_id')
+
+                cache_key = (f"{platform_metadata.get('org_id')}"
+                             f"{CACHE_KEYWORD_FOR_DELETED_SYSTEM}{host['id']}")
+                if cache.get(cache_key):
+                    LOG.info(
+                        f"{self.prefix} - Received a msg for deleted system "
+                        f" with inventory id: {host['id']}.Hence, rejecting a msg."
+                    )
+                    return
```

Comment:
I think before return we need to unset the cache? else when there is new incoming message and cache is still intact engine processor would deny the entry till  `CACHE_TIMEOUT_FOR_DELETED_SYSTEM`? Though it seems there is very little chance that org_id and host['id'] combination will repeat to make this happen?

Reply:
+1 Good point @upadhyeammit! I have added delete cache call after getting create/update event for same system. So that it will process incoming messages for the same system.